sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,268 @@
+    G3 = TransferFunction(p, p**4 - 6, p)
+    G4 = TransferFunction(p + 4, p - 5, p)
+    G5 = TransferFunction(s + 6, s - 5, s)
+    G6 = TransferFunction(s + 3, s + 1, s)
+
+    # multiplication
+    expect1 = TransferFunction((s + 1)*(s + 3), (9 - s**3)*(s - 5), s)
+    assert G1*G2 == expect1
+
+    expect2 = TransferFunction(p*(p + 4), (p - 5)*(p**4 - 6), p)
+    assert G3*G4 == expect2
+    assert G5*(s - 1) == TransferFunction((s - 1)*(s + 6), s - 5, s)
+    assert 9*G5 == TransferFunction(9*s + 54, s - 5, s)
+
+    c = symbols("c", commutative=False)
+    raises(ValueError, lambda: G3 * Matrix([1, 2, 3]))
+    raises(ValueError, lambda: G1 * c)
+    raises(ValueError, lambda: G3 * G5)
+
+    # division
+    assert G5/G6 == TransferFunction((s + 1)*(s + 6), (s - 5)*(s + 3), s)
+    assert G5/2 == TransferFunction(s + 6, 2*s - 10, s)
+    assert G5/(s**2) == TransferFunction(s + 6, s**2*(s - 5), s)
+    assert (s - 4*s**2)/G2 == TransferFunction((s - 5)*(-4*s**2 + s), s + 1, s)
+    assert 0/G4 == TransferFunction(0, p + 4, p)
+
+    raises(ValueError, lambda: G3 / Matrix([1, 2, 3]))
+    raises(ValueError, lambda: G6 / 0)
+    raises(ValueError, lambda: G3 / G5)
+
+    p1 = (9 - s**3)*(s - 5)*(s + 1)*(s + 6) + (s - 5)*(s + 1)*(s + 3)**2
+    p2 = (9 - s**3)*(s - 5)**2*(s + 3)
+    assert G1*G2 + G5/G6 == TransferFunction(p1, p2, s)
+
+    G3_ = TransferFunction(s**4 - 2*s**2 + 1, s - 1, s)
+    p3 = (s - 5)*(s - 1)*(s + 3)
+    p4 = (9 - s**3)*(s + 1)*(s**4 - 2*s**2 + 1)
+    assert G1/G2/G3_ == TransferFunction(p3, p4, s)
+
+    p5 = (s - 5)*(s - 1)*(s + 3)**2 - (9 - s**3)*(s + 1)**2*(s**4 - 2*s**2 + 1)
+    p6 = (9 - s**3)*(s - 1)*(s + 1)*(s + 3)
+    assert G1/G2 - G3_/G6 == TransferFunction(p5, p6, s)
+
+
+def test_TransferFunction_is_proper():
+    G1 = TransferFunction(s**4 + s**3 - 2*s, s**4 - 1, s)
+    G2 = TransferFunction(p**7, p - 4, p)
+    G3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    G4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert G1.is_proper
+    assert not G2.is_proper
+    assert G3.is_proper
+    assert not G4.is_proper
+
+
+def test_TransferFunction_is_strictly_proper():
+    tf1 = TransferFunction(s**3 - 2, s**4 + 5*s + 6, s)
+    tf2 = TransferFunction(p + 3, p - 4, p)
+    tf3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    tf4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert tf1.is_strictly_proper
+    assert not tf2.is_strictly_proper
+    assert tf3.is_strictly_proper
+    assert not tf4.is_strictly_proper
+
+
+def test_TransferFunction_is_biproper():
+    tf1 = TransferFunction(s - 1, s - 2, s)
+    tf2 = TransferFunction(p**3 - 2, p**4 + 5*p + 6, p)
+    tf3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    tf4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert tf1.is_biproper
+    assert not tf2.is_biproper
+    assert not tf3.is_biproper
+    assert not tf4.is_biproper
```

Comment:
I want to reiterate that numbers are used instead of Symbol() for the vast majority of these tests. The tests need to be written primarily as fully symbolic tests. If the symbolic tests work, then it is likely that an identical set of tests that use all numbers except for `TransferFunction.var` will work. We are missing checks for a lot of bugs. I don't see a single test that has a symbolic polynomial coefficient, for example.